source/tests/unit/calibration_operations_tests.cpp

```diff
@@ -0,0 +1,638 @@
+    NISysCfgBoolTrue, //internal_calibration_supported
+    3, //number_of_internal_calibration_regions
+    NISysCfgBoolTrue, //external_calibration_supported
+    kExtCalLastTemp1,
+    kExtCalLastTime1,
+    kNextCalTime1);
+
+  EXPECT_CALL(*mock_library, GetResourceIndexedProperty(_, NISysCfgIndexedPropertyInternalCalibrationName, _, _))
+      .WillOnce(Return(NISysCfg_PropDoesNotExist));
+  EXPECT_CALL(*mock_library, GetResourceIndexedProperty(_, NISysCfgIndexedPropertyInternalCalibrationLastTemp, _, _))
+      .WillOnce(Return(NISysCfg_PropDoesNotExist));
+  EXPECT_CALL(*mock_library, GetResourceIndexedProperty(_, NISysCfgIndexedPropertyInternalCalibrationLastTime, _, _))
+      .WillOnce(Return(NISysCfg_PropDoesNotExist));
+}
+
+TEST(CalibrationOperationsTests, SupportsIntCal_GetCalibrationInformation_ResponseWritten)
+{
+  NiceMock<ni::tests::utilities::SysCfgMockLibrary> mock_library;
+  nidevice_restricted_grpc::CalibrationOperationsRestrictedService service(&mock_library);
+  ::grpc::ServerContext context;
+  nidevice_restricted_grpc::GetCalibrationInformationRequest request;
+  nidevice_restricted_grpc::GetCalibrationInformationResponse response;
+  configure_mock_library_one_internal_calibration_region(&mock_library);
+
+  ::grpc::Status status = service.GetCalibrationInformation(&context, &request, &response);
+
+  EXPECT_TRUE(status.ok());
+  EXPECT_TRUE(response.supports_internal_calibration());
+  EXPECT_TRUE(response.supports_internal_calibration_available());
+}
+
+TEST(CalibrationOperationsTests, DoesNotSupportIntCal_GetCalibrationInformation_ResponseWritten)
+{
+  NiceMock<ni::tests::utilities::SysCfgMockLibrary> mock_library;
+  nidevice_restricted_grpc::CalibrationOperationsRestrictedService service(&mock_library);
+  ::grpc::ServerContext context;
+  nidevice_restricted_grpc::GetCalibrationInformationRequest request;
+  nidevice_restricted_grpc::GetCalibrationInformationResponse response;
+  configure_mock_library_no_calibration_support(&mock_library);
+
+  ::grpc::Status status = service.GetCalibrationInformation(&context, &request, &response);
+
+  EXPECT_TRUE(status.ok());
+  EXPECT_FALSE(response.supports_internal_calibration());
+  EXPECT_TRUE(response.supports_internal_calibration_available());
+}
+
+TEST(CalibrationOperationsTests, SupportsIntCalPropDoesNotExist_GetCalibrationInformation_ResponseWritten)
+{
+  NiceMock<ni::tests::utilities::SysCfgMockLibrary> mock_library;
+  nidevice_restricted_grpc::CalibrationOperationsRestrictedService service(&mock_library);
+  ::grpc::ServerContext context;
+  nidevice_restricted_grpc::GetCalibrationInformationRequest request;
+  nidevice_restricted_grpc::GetCalibrationInformationResponse response;
+  configure_mock_library_calibration_properties_do_not_exist(&mock_library);
+
+  ::grpc::Status status = service.GetCalibrationInformation(&context, &request, &response);
+
+  EXPECT_TRUE(status.ok());
+  EXPECT_FALSE(response.supports_internal_calibration_available());
```

Comment:
Is there any value in asserting on the defaults set for the actual field in `CalibrationOperationsRestrictedService::GetCalibrationInformation` even when the `_available` field is false? I don't imagine them changing though and I'm assuming they're all correct in that file (False for bools, 0 for numerics, and empty for repeated fields) so may not be any value adding them in the tests.

But thought I'd call it out since the singular fields are explicitly being set in the service if it's available or not.

Reply:
I'll revisit these when I change the fields to `optional`.

Reply:
Added checks for default values.